imagetest/testworkflow.go

```diff
@@ -216,9 +216,14 @@ func (t *TestWorkflow) addWaitStep(stepname, vmname string, stopped bool) (*dais
 	instanceSignal.Name = vmname
 	instanceSignal.Stopped = stopped
 
+	guestAttribute := &daisy.GuestAttribute{}
+	guestAttribute.Namespace = utils.GuestAttributeTestNamespace
+	guestAttribute.KeyName = utils.GuestAttributeTestKey
+
 	// Waiting for stop and waiting for success match are mutually exclusive.
 	if !stopped {
 		instanceSignal.SerialOutput = serialOutput
+		instanceSignal.GuestAttribute = guestAttribute
```

Comment:
This will wait for both the stopped signal and the test-complete key when stopped is true. I don't think this is the right pattern of behavior, when `addWaitStep` is called in fixtures.go with `stopped=true` it's called as a part of rebooting where the instance stopping is the critical signal rather than the tests being finished.
Maybe this should be broken out into two different functions with clearer meanings (ie wait for a stopped VM vs wait for tests to be finished).

Reply:
The new change has a waitstep and a waitstoppedstep, where the SerialOutput and GuestAttrbute are only checked in the waitstep case.

It appears that if both are set, the wait step finishes if either the serial log or guest attribute matches.

Reply:
Ok so it will still work for reboots, but then it won't correctly signal when tests are finished, right?

Reply:
Nevermind I see you fixed it in the latest commit.

Reply:
i haven't tested all the suites which call reboot(), though when I ran the imageboot suite, which calls reboot(), the test suite completed successfully.